File: Manual/test-cases/examples/chill/unroll.c
#define N 14
void foo(int n, float* x, float* y, float* z, float* f3, float* f1, float* w) {
	int dt;

	int i, j;

	for (i = 1; i <= 14; i++)
		x[i] = 1.0;

	for (i = 1; i <= 14; i += 3)
		y[i] = 1.0;

	for (i = N + 1; i <= N + 20; i += 3)
		z[i] = 1.0;

	for (i = 0; i <= N; i++) {
		for (j = i; j <= i + N; j++)
			f3[i] = f3[i] + f1[j] * w[j - i];
		f3[i] = f3[i] * dt;
	}

	return 0;
}

int main() {
	float x[N], y[N], z[N], f3[N], f1[N], w[N];

	foo(N, x, y, z, f3, f1, w);
	return 0;
}

File: examples/chill/unroll.f
      SUBROUTINE foo(N,X,Y,Z,F3,F1,W)
      INTEGER N
      REAL*8  X(N),Y(N),Z(N),F3(0:N),F1(0:N),W(0:N)
      REAL*8 DT
      
      INTEGER I,J


       DO 10,I=1,14
         X(I) = 1.0D0
 10   CONTINUE
    
      DO 20,I=1,14,3
         Y(I) = 1.0D0
 20   CONTINUE

       DO 30,I=N+1,N+20,3
         Z(I) = 1.0D0
 30   CONTINUE
    
      DO 40,I=0,N
         DO 50,J=I,I+N
            F3(I) = F3(I) + F1(J) * W(J-I)
 50      CONTINUE
         F3(I) = F3(I) * DT
 40   CONTINUE

      END
